src/features/envVisualizer/components/values/UnassignedValue.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import { Config } from '../../EnvVisualizerConfig';
+import { Layout } from '../../EnvVisualizerLayout';
+import { ReferenceType, UnassignedData } from '../../EnvVisualizerTypes';
+import { getTextWidth } from '../../EnvVisualizerUtils';
+import { Binding } from '../Binding';
+import { Text } from '../Text';
+import { Value } from './Value';
+
+/** this class encapsulates an unassigned value in Source, internally represented as a symbol */
+export class UnassignedValue extends Value {
+  readonly x: number;
+  readonly y: number;
+  readonly height: number;
+  readonly width: number;
+  readonly data: UnassignedData = Symbol();
+  readonly text: Text;
+
+  constructor(readonly referencedBy: ReferenceType[]) {
+    super();
+
+    // derive the coordinates from the main reference (binding / array unit)
+    const mainReference = this.referencedBy[0];
+    if (mainReference instanceof Binding) {
+      this.x = mainReference.x + getTextWidth(mainReference.keyString) + Config.TextPaddingX;
+      this.y = mainReference.y;
+      this.text = new Text('', this.x, this.y, { isStringIdentifiable: false });
```

Comment:
new Text(Config.UnassignedData.toString(), this.x, this.y, { isStringIdentifiable: false });